lib/ngx/re.lua

```diff
@@ -199,7 +199,9 @@ function _M.split(subj, regex, opts, ctx, max, res)
 
                 sub_idx = to
 
-                if sub_idx >= len then
+                if sub_idx > len then
+                    break
+                elseif sub_idx == len and last_empty_match then
```

Comment:
I tested this locally and this patch seems unnecessary,  this code path (with max) seems unaffected by this behavior, and succeeds without this patch.

Reply:
test case like re-split.t TEST 39:

``` lua
local ngx_re = require "ngx.re"
local res, err = ngx_re.split("a,b,cd,,,", ",", 6)

ngx.say(#res)  
-- output is 5
-- it shoule be 6 here

ngx.say(table.concat(res, "|")) 
-- outout is a|b|cd||, 
-- it should be a|b|cd|||
```

perl's behavior
``` bash
perl -le 'print join "|", split ",", "a,b,cd,,,", 6'
a|b|cd|||
```

Reply:
Apologies, I had the following patch when testing this locally and failed to spot the difference with diff here:
```diff
diff --git a/lib/ngx/re.lua b/lib/ngx/re.lua
index 06b5e9b..6ccc958 100644
--- a/lib/ngx/re.lua
+++ b/lib/ngx/re.lua
@@ -199,11 +199,17 @@ function _M.split(subj, regex, opts, ctx, max, res)

                 sub_idx = to

-                if sub_idx >= len then
+                if sub_idx > len then
                     break
                 end
```

This patch suffices to make TEST 38 and 39 pass :)

Reply:
:)